ydb/core/tx/schemeshard/schemeshard__table_stats.cpp

```diff
@@ -137,6 +157,12 @@ TPartitionStats TTxStoreTableStats::PrepareStats(const TActorContext& ctx, const
     newStats.IndexSize = tableStats.GetIndexSize();
     newStats.LastAccessTime = TInstant::MilliSeconds(tableStats.GetLastAccessTime());
     newStats.LastUpdateTime = TInstant::MilliSeconds(tableStats.GetLastUpdateTime());
+    for (const auto& channelStats : tableStats.GetChannels()) {
+        const auto& poolKind = channelsMapping.at(channelStats.GetChannel());
```

Comment:
@snaury Could channelStats.GetChannel() be different from the index of the channel in the tableStats.GetChannels()? If so, could .GetChannel() be out of bounds of the Channels array?

Reply:
Probably not, we don't support removing channels, so channel bindings may grow over time, but never shrink. However please note that new schemeshard generation may execute an alter that reconfigures channels, and datashard may theoretically connect to older schemeshard generation which doesn't know about new channels. Please handle any mismatches gracefully without crashing.

Reply:
@snaury Could it be that the channels of a datashard are reconfigured in such a way that the storage pool will have the same name as before, but a different kind? Or could it happen that the storage pool name is also changed in TChannelsBindings, but to some other storage pool that has a different kind than this channel had before?

Reply:
Discussed this privately with @snaury and I believe the answer is yes, channels can be reconfigured to use a different storage pool kind. However, currently nobody does this and channels are only added if a new configuration is needed, not changed.